x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeParseProcessor.java

```diff
@@ -79,12 +111,12 @@ public String getWriteableName() {
 
     @Override
     protected Object doProcess(Object timestamp, Object pattern) {
-        return process(timestamp, pattern, zoneId());
+        return this.parser.parse(timestamp, pattern, zoneId());
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(left(), right());
+        return Objects.hash(parser, left(), right());
```

Comment:
```suggestion
        return Objects.hash(left(), right(), parser);
```

Reply:
What's the difference?

Reply:
Minor, just to have the same order as in the constructor.